apps/omg_api/lib/fee_checker/core.ex

```diff
@@ -18,29 +18,11 @@ defmodule OMG.API.FeeChecker.Core do
   """
 
   alias OMG.API.Crypto
-  alias OMG.API.State.Transaction
-  alias OMG.API.State.Transaction.Recovered
   alias Poison
 
   @type fee_spec_t() :: %{token: Crypto.address_t(), flat_fee: non_neg_integer}
   @type token_fee_t() :: %{Crypto.address_t() => non_neg_integer}
 
-  @doc """
-  Calculates fee from transaction and checks whether token is allowed and flat fee limits are met
-  """
-  @spec transaction_fees(Recovered.t(), token_fee_t()) :: {:ok, token_fee_t()} | {:error, :token_not_allowed}
-  def transaction_fees(
-        %Recovered{signed_tx: %Transaction.Signed{raw_tx: tx}},
-        token_fees
-      ) do
-    currencies = Transaction.get_currencies(tx)
-    tx_fees = Map.take(token_fees, currencies)
```

Comment:
I'm ok with keeping simple, but let's make a mental note that passing the whole fee map around might need to be optimized in the future (which is what, among other things, this function ensured by that `Map.take`)

Alternatively you can make a step back and just ditch the latter `if`